src/v/kafka/server/group_manager.cc

```diff
@@ -1802,6 +1804,35 @@ group_manager::describe_partition_producers(const model::ntp& ntp) {
     return response;
 }
 
+ss::future<std::error_code> group_manager::empty_and_delete_groups(
+  const model::ntp& ntp, const chunked_vector<group_id>& groups) {
+    co_await ssx::async_for_each(groups, [this, &ntp](const group_id& group) {
+        auto g = get_group(group);
+        if (!g) {
+            vlog(cg_klog.warn, "Group {} not found on ntp {}", group, ntp);
+            return;
+        }
+        g->remove_full_members();
+    });
+
+    std::vector<std::pair<model::ntp, group_id>> groups_with_ntps{
```

Comment:
should we use chunked vector, or is this intended to be small ? i guess per group vector may grow pretty large